### PR TITLE
#167190435 User can reset password

### PR DESCRIPTION
**Pull request by encodedBicoding**
#### What does this PR do?
- creates an endpoint that collects a user email, and send that email a reset link
- upon clicking the reset link, the user adds the new password and the user password is changed
#### Description of Task to be completed?
- writes test.
- creates a new table migration named PasswordResetTokens and save the tokens generated in there
- checks the token availability with that of what was sent to the user
- does action according to the token availability
- document on swagger
#### How should this be manually tested?
- pull this branch or fetch it
- run `npm run dev`
- use swagger docs to test the endpoint on `http://localhost:3000/api/v1/api-docs` or click [link](http://localhost:3000/api/v1/api-docs)
#### Any background context you want to provide?
- This method of validating token, where I created a new table migration, is bogus for small-scale projects, but it needed when working on a large scale project because data is invaluable
#### What are the relevant pivotal tracker stories?
[#167190435](https://www.pivotaltracker.com/story/show/167190435)
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "prettier.eslintIntegration": true
+}
```
Comment (chuxmykel) [inline .vscode/settings.json]: I can see this file in the `.gitignore`. I think you committed it before adding it to the `.gitignore`. Try deleting it and making a new commit.

File: package.json [hunk 2]
```diff
@@ -11,17 +11,22 @@
...
+    "seed": "npm run seed:all",
+    "seed:all": "node_modules/.bin/sequelize db:seed:all",
+    "seed:undo": "node_modules/.bin/sequelize db:seed:undo:all",
     "generate:model": "node_modules/.bin/sequelize model:generate",
     "migrate": "node_modules/.bin/sequelize db:migrate",
     "migrate:undo": "node_modules/.bin/sequelize db:migrate:undo:all",
     "coveralls": "nyc report --reporter=text-lcov | coveralls",
-    "copy-docs": "cp -r server/docs/ build/docs/"
+    "copy-docs": "cp -r server/docs/ build/docs/",
+    "testmail": "export DEBUG=dev && nodemon --exec babel-node server/helpers/mail/mail"
```
Comment (chuxmykel) [inline package.json]: Looks like you synced with @cvjude on this one. Nice one guys.

**Review by Hadeneekeh — COMMENTED** [review]
Well done @encodedBicoding! Can you look into Hound's comment?

**Review by Lundii — APPROVED** [review]
Works fine on my local. Great job

**Review by Hadeneekeh — APPROVED** [review]

**Review by chuxmykel — APPROVED** [review]
LGTM

**Review by chiboycalix — APPROVED** [review]
LGTM

**Review by topseySuave — APPROVED** [review]

**Pull request merged**